src/coffee/client/modules/todo/todo_item_module.coffee

```diff
@@ -0,0 +1,69 @@
+#
+# Copyright (C) 2015 by Looker Data Services, Inc.
+# All rights reserved.
+#
+
+angular   = require 'angular'
+templates = require '../../templates'
+{EVENT}   = require '../../../constants'
+_         = require '../../../underscore'
+
+############################################################################################################
+
+### ex: TodoItem flow
+1. User clicks the submit button, `$scope.addTodoItem()` fires TodoItems's addTodoItem action
+2. `onAddItem()` runs and creates a new ReadOnlyView TodoItem, pushes it into @_items
+    (3. Register doItem action to run when the 'done' field of the ReadOnlyView is changed)
+4. TodoItemStore sends a notification that something has been added and changed
+
+############################################################################################################
+
+todo = angular.module('todo', ['reflux'])
+
+todo.factory 'TodoItemStore', (Todo, TodoItemActions, TextInputStore, ReadOnlyView, reflux) ->
+    reflux.createStore
+
+        listenables: TodoItemActions
+
+        init: ->
+            @_items = []
+
+        onAddItem: -> # (2)
+            @_todo = new ReadOnlyView({
+                id: _.uniqueId 'todo-'
+                description: TextInputStore._value
+                done: false
+            }, ['id', 'description', 'done'])
+            @_items.unshift(@_todo)
+
+            @_todo.setAction('done', TodoItemActions.doItem) # (3)
+
+            @trigger(EVENT.ADD, @_todo.id) # (4)
+            @trigger(EVENT.CHANGE) # (4)
+
+        onRemoveItem: (id) ->
+            @_items = _.filter @_items, (item) -> item.id isnt id
```

Comment:
An alternative implementation is to use a loop and the `splice` method:

``` coffeescript
for item, index in @_items
    continue unless item.id is id
    @_items.splice index, 1
    break
```

The major difference is that the current code will create a brand new array while the alternative does not (but still needs to re-index the array).  Depending upon the circumstances, you might need one instead of the other.  My default is to use `splice` as it avoids creating new objects and giving the garbage collector more work.